Neos.ContentGraph.DoctrineDbalAdapter/src/ContentGraphTableNames.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Neos\ContentGraph\DoctrineDbalAdapter;
+
+/**
+ * Encapsulates table name generation for content graph tables
+ * @internal
+ */
+final class ContentGraphTableNames
```

Comment:
Just wondering: In other implementations we just pass the `$tableNamePrefix` through and instead of:
```php
new Table($this->contentGraphTableNames->node(), ...
```
do something like
```php
new Table($this->tableNamePrefix . '_node', ...
```

Not a rhetoric question: do you think, it makes things easier to introduce a class for this? Because if so, we should maybe adjust other projections too

Reply:
I also started adjusting what bastian noted and i also thought: Why always recreate the helper instance instead of passing it along:

https://github.com/kitsunet/neos-development-collection/pull/3

But it seems there are still a LOT of places using the legacy string syntax - for example the content graph projection will even use both ways currently (id 'deprecated' the second):

```
private readonly ContentGraphTableNames $contentGraphTableNames;
/** @deprecated */
private readonly string $tableNamePrefix;
```

Also the `NodeRecord` and `HierarchyRelation` active records mix the style (by using the new `DimensionSpacePointsRepository`)

Additionally i would definitely prefer a shortened name like `$tableNames` to simplify readably (we know we are scoped to the graph)

In the end im not against this refactoring - though it should have probably been separated ^^ - but it should be carried out consistently. Ill try to see if i can easily adjust the leftover cases tomorrow (see my pr).

Reply:
Okay i refactored all the things in my pr. And found a typo `ContentGraphTableNames::hierachyRelation` vs `hierarchyRelation` :D

Reply:
Last todo is to adjust the projection integrity violation thingy and make $tablePrefixName private